vale/code/arch/ppc64le/Vale.PPC64LE.InsBasic.vaf

```diff
@@ -0,0 +1,329 @@
+var r1:nat64 {:state reg(1)};
+var r2:nat64 {:state reg(2)};
+var r3:nat64 {:state reg(3)};
+var r4:nat64 {:state reg(4)};
+var r5:nat64 {:state reg(5)};
+var r6:nat64 {:state reg(6)};
+var r7:nat64 {:state reg(7)};
+var r8:nat64 {:state reg(8)};
+var r9:nat64 {:state reg(9)};
+var r10:nat64 {:state reg(10)};
+var r11:nat64 {:state reg(11)};
+var r12:nat64 {:state reg(12)};
+var r13:nat64 {:state reg(13)};
+var r14:nat64 {:state reg(14)};
+var r15:nat64 {:state reg(15)};
+var r16:nat64 {:state reg(16)};
+var r17:nat64 {:state reg(17)};
+var r18:nat64 {:state reg(18)};
+var r19:nat64 {:state reg(19)};
+var r20:nat64 {:state reg(20)};
+var r21:nat64 {:state reg(21)};
+var r22:nat64 {:state reg(22)};
+var r23:nat64 {:state reg(23)};
+var r24:nat64 {:state reg(24)};
+var r25:nat64 {:state reg(25)};
+var r26:nat64 {:state reg(26)};
+var r27:nat64 {:state reg(27)};
+var r28:nat64 {:state reg(28)};
+var r29:nat64 {:state reg(29)};
+var r30:nat64 {:state reg(30)};
+var r31:nat64 {:state reg(31)};
+var cr0:cr0_t {:state cr0()};
+var xer:xer_t {:state xer()};
+var mem_config:int {:state mem_config()};
+var mem:vale_heap {:state mem()};
+var heap0 :vale_heap {:state mem_heaplet(0 )}{:updates mem};
+var heap1 :vale_heap {:state mem_heaplet(1 )}{:updates mem};
+var heap2 :vale_heap {:state mem_heaplet(2 )}{:updates mem};
+var heap3 :vale_heap {:state mem_heaplet(3 )}{:updates mem};
+var heap4 :vale_heap {:state mem_heaplet(4 )}{:updates mem};
+var heap5 :vale_heap {:state mem_heaplet(5 )}{:updates mem};
+var heap6 :vale_heap {:state mem_heaplet(6 )}{:updates mem};
+var heap7 :vale_heap {:state mem_heaplet(7 )}{:updates mem};
+var heap8 :vale_heap {:state mem_heaplet(8 )}{:updates mem};
+var heap9 :vale_heap {:state mem_heaplet(9 )}{:updates mem};
+var heap10:vale_heap {:state mem_heaplet(10)}{:updates mem};
+var heap11:vale_heap {:state mem_heaplet(11)}{:updates mem};
+var heap12:vale_heap {:state mem_heaplet(12)}{:updates mem};
+var heap13:vale_heap {:state mem_heaplet(13)}{:updates mem};
+var heap14:vale_heap {:state mem_heaplet(14)}{:updates mem};
+var heap15:vale_heap {:state mem_heaplet(15)}{:updates mem};
+var stack:vale_stack {:state stack()};
+var memLayout:vale_heap_layout {:state mem_layout()};
+var stackTaint:memtaint {:state stackTaint()};
+
+// Operands of general-purpose registers
+operand_type reg_opr:nat64 @ reg :=
+| inout r0 | inout r1 | inout r2 | inout r3
+| inout r4 | inout r5 | inout r6 | inout r7
+| inout r8 | inout r9 | inout r10 | inout r11
+| inout r12 | inout r13 | inout r14 | inout r15
+| inout r16 | inout r17 | inout r18 | inout r19
+| inout r20 | inout r21 | inout r22 | inout r23
+| inout r24 | inout r25 | inout r26 | inout r27
+| inout r28 | inout r29 | inout r30 | inout r31
+;
+// 64-bit Memory operand
+operand_type Mem64(in h:heaplet, in base:reg64, inline offset:int, ghost b:buffer64, ghost index:int, inline t:taint):int;
+operand_type Stack(in base:reg_opr, inline offset:int, inline t:taint):int;
```

Comment:
On a RISC architecture, do you need Mem64 and Stack operands?  If not, you could just omit these operand_types, along with the Mem64_in, Mem64_lemma, Stack_in, Stack_lemma.

Reply:
Some load/store instructions have the form of `offset(base_reg)` for memory operands, that's why I kept these operand_types around. But since I flattened that form into two operands in all recurrences for simpleness, it makes sense to get rid of them.